src/entity-schema/EntitySchemaEmbeddedOptions.ts

```diff
@@ -0,0 +1,19 @@
+import { EntitySchema } from "./EntitySchema";
+
+export interface EntitySchemaEmbeddedOptions {
+    /**
+     * Type of the class to be embedded.
+     */
+    type: ((type?: any) => Function | EntitySchema);
```

Comment:
I think here we should point to either Function either entity schema name, e.g. `Function | string`, don't really want to have a circular deps within entity schemas too.